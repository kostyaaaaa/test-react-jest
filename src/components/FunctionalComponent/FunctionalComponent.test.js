import Component from "./";
import { mount } from "enzyme";

let wrapper;

describe("FunctionalComponent", () => {
  beforeEach(() => {
    wrapper = mount(<Component></Component>);
  });
  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).not.toBeNull();
  });
  it("should render 0 count by default", () => {
    expect(wrapper.find(".count").text()).toEqual("0");
  });
  it("should increase count when button click", () => {
    wrapper.find("button").simulate("click");
    expect(wrapper.find(".count").text()).toEqual("1");
  });
  it("should update input value when button clicked twice", () => {
    wrapper.find("button").simulate("click");
    wrapper.find("button").simulate("click");
    wrapper.find("button").simulate("click");
    expect(wrapper.find("input").get(0).props.value).toEqual(3);
    expect(wrapper.find("input").instance().value).toEqual("3");
  });
  it("should update text when type in input", () => {
    wrapper.find("input").simulate("change", { target: { value: "123" } });
    expect(wrapper.find("input").get(0).props.value).toEqual("123");
  });
});
