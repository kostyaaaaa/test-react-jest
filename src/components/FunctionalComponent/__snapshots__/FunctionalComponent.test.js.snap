// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FunctionalComponent should render correctly 1`] = `ReactWrapper {}`;
